pkg/eval/glob.go

```diff
@@ -75,6 +89,24 @@ func (gp GlobPattern) Index(k interface{}) (interface{}, error) {
 		gp.Segments[len(gp.Segments)-1] = glob.Wild{
 			lastSeg.Type, true, lastSeg.Matchers,
 		}
+	case modifier == "d" || modifier == "r" || strings.HasPrefix(modifier, "type:"):
+		// This handles cases such as `**x*[r][d]`.
+		if gp.TypeCb != nil {
+			return nil, ErrMultipleTypeModifiers
+		}
+		switch {
+		case modifier == "d":
+			gp.TypeCb = typeMap[typeDir]
+		case modifier == "r":
+			gp.TypeCb = typeMap[typeRegular]
+		default:
+			ptype := modifier[len("type:"):]
```

Comment:
I know you can't use `type` since it's a keyword, but not clear what `ptype` means; `typeName` is better. Or just don't use a variable here, it is only used in the next statement.

Reply:
Renamed. Note: I prefer a temp var in this situation because it's much clearer to a human and shouldn't be any slower.